Editor configs

yorinasub17 opened the pull request:
cc @rameshvs 

This adds controls to turn on/off the following two options in the query editor:
- Vim bindings
- Live autocompletion (autocompletion as you type)

Also adds CSS styles to extend the autocompletion window.

yorinasub17 committed:
Editor configs

woohp commented on client/app/pages/query-snippets/edit.js:
```diff
@@ -16,6 +24,24 @@ function SnippetCtrl($routeParams, $http, $location, toastr, currentUser, Events
       editor.getSession().setUseWrapMode(true);
       editor.setShowPrintMargin(false);
+
+      $scope.$watch(
```
Instead of watching a variable, you can use `ng-change` in the template to explicitly trigger.
```
<input ... ng-change="changeVimMode($event)">
```

yorinasub17 replied:
If I do that, I can't access the editor...

yorinasub17 commented:
@woohp I addressed your comment. I took out the liveautocompletion toggle from the snippets view, because redash didn't actually have autocompletion enabled on that view so the toggle was meaningless.
